code/modules/research/rdconsole.dm

```diff
@@ -71,69 +58,114 @@ won't update every console in existence) but it's more of a hassle to do. Also,
-/obj/machinery/computer/rdconsole/proc/griefProtection()
-	for(var/obj/machinery/r_n_d/server/centcom/C in GLOB.machines)
-		for(var/v in files.known_tech)
-			var/datum/tech/T = files.known_tech[v]
-			C.files.AddTech2Known(T)
-		for(var/v in files.known_designs)
-			var/datum/design/D = files.known_designs[v]
-			C.files.AddDesign2Known(D)
-		C.files.RefreshResearch()
-
 
 /obj/machinery/computer/rdconsole/Initialize()
 	. = ..()
-	files = new /datum/research(src) //Setup the research data holder.
+	stored_research = SSresearch.science_tech
+	stored_research.consoles_accessing[src] = TRUE
 	matching_designs = list()
-	if(!id)
-		fix_noid_research_servers()
+	SyncRDevices()
 
-/*	Instead of calling this every tick, it is only being called when needed
-/obj/machinery/computer/rdconsole/process()
-	griefProtection()
-*/
+/obj/machinery/computer/rdconsole/Destroy()
+	if(stored_research)
+		stored_research.consoles_accessing -= src
+	if(linked_destroy)
+		linked_destroy.linked_console = null
+		linked_destroy = null
+	if(linked_lathe)
+		linked_lathe.linked_console = null
+		linked_lathe = null
+	if(linked_imprinter)
+		linked_imprinter.linked_console = null
+		linked_imprinter = null
+	if(t_disk)
+		t_disk.forceMove(get_turf(src))
+		t_disk = null
+	if(d_disk)
+		d_disk.forceMove(get_turf(src))
+		d_disk = null
+	matching_designs = null
+	selected_node = null
+	selected_design = null
+	return ..()
 
 /obj/machinery/computer/rdconsole/attackby(obj/item/D, mob/user, params)
-
 	//Loading a disk into it.
 	if(istype(D, /obj/item/disk))
-		if(t_disk || d_disk)
-			to_chat(user, "A disk is already loaded into the machine.")
-			return
-
 		if(istype(D, /obj/item/disk/tech_disk))
+			if(t_disk)
+				to_chat(user, "<span class='danger'>A technology disk is already loaded!</span>")
+				return
+			if(!user.transferItemToLoc(D, src))
+				to_chat(user, "<span class='danger'>[D] is stuck to your hand!</span>")
+				return
 			t_disk = D
 		else if (istype(D, /obj/item/disk/design_disk))
+			if(d_disk)
+				to_chat(user, "<span class='danger'>A design disk is already loaded!</span>")
+				return
+			if(!user.transferItemToLoc(D, src))
+				to_chat(user, "<span class='danger'>[D] is stuck to your hand!</span>")
+				return
 			d_disk = D
 		else
 			to_chat(user, "<span class='danger'>Machine cannot accept disks in that format.</span>")
 			return
-		if(!user.transferItemToLoc(D, src))
-			return
-		to_chat(user, "<span class='notice'>You add the disk to the machine!</span>")
+		to_chat(user, "<span class='notice'>You insert [D] into \the [src]!</span>")
 	else if(!(linked_destroy && linked_destroy.busy) && !(linked_lathe && linked_lathe.busy) && !(linked_imprinter && linked_imprinter.busy))
 		. = ..()
-	updateUsrDialog()
 
+/obj/machinery/computer/rdconsole/proc/research_node(id, mob/user)
+	if(!stored_research.available_nodes[id] || stored_research.researched_nodes[id])
+		say("Node unlock failed: Either already researched or not available!")

```

Comment:
These should probably be alerts on the UI instead but i guess that can be changed later.